norse/torch/module/receptive_field.py

```diff
@@ -4,85 +4,260 @@
+from typing import Callable, NamedTuple, Optional, Tuple
 
 import torch
 
-from norse.torch.module.leaky_integrator_box import LIBoxCell, LIBoxParameters
 from norse.torch.module.snn import SNNCell
+from norse.torch.module.leaky_integrator_box import LIBoxCell, LIBoxParameters
 from norse.torch.functional.receptive_field import (
     spatial_receptive_fields_with_derivatives,
-    temporal_scale_distribution,
+    spatial_parameters,
+    temporal_scale_distribution
 )
 
 
 class SpatialReceptiveField2d(torch.nn.Module):
     """Creates a spatial receptive field as 2-dimensional convolutions.
-    The parameters decide the number of combinations to scan over, i. e. the number of receptive fields to generate.
-    Specifically, we generate ``n_scales * n_angles * (n_ratios - 1) + n_scales`` output_channels with aggregation,
-    and ``in_channels * (n_scales * n_angles * (n_ratios - 1) + n_scales)`` without aggregation.
+    The `rf_parameters` are a tensor of shape `(n, 5)` where `n` is the number of receptive fields.
+    If the `optimize_fields` flag is set to `True`, the `rf_parameters` will be optimized during training.
 
-    The ``(n_ratios - 1) + n_scales`` terms exist because at ``ratio = 1``, fields are perfectly symmetrical, and there
-    is therefore no reason to scan over the angles and scales for ``ratio = 1``.
-    However, ``n_scales`` receptive field still needs to be added (one for each scale-space).
+    Example:
+        >>> import torch
+        >>> from norse.torch import SpatialReceptiveField2d
+        >>> parameters = torch.tensor([[1., 1., 1., 0., 0.]])
+        >>> m = SpatialReceptiveField2d(1, 9, parameters)
+        >>> m.weights.shape
+        torch.Size([1, 1, 9, 9])
+        >>> y = m(torch.empty(1, 1, 9, 9))
+        >>> y.shape
+        torch.Size([1, 1, 1, 1])
 
-    Parameters:
-        n_scales (int): Number of scaling combinations (the size of the receptive field) drawn from a logarithmic distribution
-        n_angles (int): Number of angular combinations (the orientation of the receptive field)
-        n_ratios (int): Number of eccentricity combinations (how "flat" the receptive field is)
-        size (int): The size of the square kernel in pixels
-        derivatives (Union[int, List[Tuple[int, int]]]): The number of derivatives to use in the receptive field.
-        aggregate (bool): If True, sums the input channels over all output channels. If False, every
-        output channel is mapped to every input channel, which may blow up in complexity.
-        **kwargs: Arguments passed on to the underlying torch.nn.Conv2d
+    Arguments:
+        in_channels (int): Number of input channels
+        size (int): Size of the receptive field
+        rf_parameters (torch.Tensor): Parameters for the receptive fields
+        aggregate (bool): If `True`, the receptive fields will be aggregated across channels. Defaults to `True`.
+        domain (float): The domain of the receptive field. Defaults to `8`.
```

Comment:
aggregate and domain parameters are not accessible from SampledSpatialReceptiveField2d and ParameterizedSpatialReceptiveField2d